src/pages.c

```diff
@@ -357,6 +357,8 @@ os_page_detect(void) {
 	SYSTEM_INFO si;
 	GetSystemInfo(&si);
 	return si.dwPageSize;
+#elif defined(__FreeBSD__)
+	return (PAGE_SIZE);
```

Comment:
Nit: jemalloc doesn't do the trailing return type thing. Otherwise LGTM.

Reply:
That should be "parenthesized return value". Brain isn't totally turned on yet.

Reply:
@trasz: this function is for sanity check of (compile time page size == runtime page size). Does the macro always reflect the runtime value on FreeBSD? I have to be honest I'm no expert there.

Reply:
Shouldn't this sanity check be disabled when built with MALLOC_PRODUCTION?  I've just tried that, and it seems it isn't.

Reply:
David, thanks for the hint.  I've amended the commit.

Reply:
We do need the check on the fly, in order to make sure the runtime value matches compile time setting (a number of internal constants like size classes are determined based on page size). This is needed because we may build on and deploy to different machines, potentially with different page sizes.

Reply:
Okay.  Still, on FreeBSD this check is not needed - page size is compile-time constant, and the sysctl simply returns that constant.  Since jemalloc and the kernel are two parts of the same source tree, changing the constant and rebuilding would still keep those two in sync.

Reply:
Well, for the built-in jemalloc, yes. However jemalloc can still be built and deployed separately, e.g. for updating to a newer version or static linking to a different version. So for strict correctness, I’m afraid the runtime check is still necessary.